### ISSUE TITLE
Tensorboard support for graphs and other features in NodeJS

### ISSUE THREAD

**Issue by thes3m** [tensorflow/tfjs#1903]
Currently Tensorboard for NodeJS only ouputs learning rate. Are there any plans to support more features than simple scalar values such as graphs, images and other features that Tensorboard supports.?

**Comment by nkreeger** [comment]
Over to @caisq who can comment more on future TensorBoard support.

**Comment by OVERSTANDING683** [comment]
Remove comp: node.js label filter #1903

**Comment by rdeprey** [comment]
I'd be interested in more Tensorboard support too. I love the Tensorflow.js library, but more features for viewing/debugging model training would be helpful.

**Comment by caisq** [comment]
Frankly, I don't think I'll personally have the cycles to fix this. So contribution will be welcome. I added tags to this issue to reflect that.

**Comment by aniket-chowdhury** [comment]
Hey, is this issue still open? If yes, how should I proceed?

**Comment by caisq** [comment]
@aniket-chowdhury Are you checking to see the state of this feature request, or are you expressing interest to contribute? This is still open. Graphs are a little tricky, because TensorFlow.js doesn't have the same computation graph underpinning as TensorFlow does. But other visualization types such as histograms and images make sense and shouldn't be highly complicated to support. I personally don't have the cycles to work on this in the near term. 

cc @kangyizhang @pyu10055 @nsthorat @dsmilkov

**Comment by aniket-chowdhury** [comment]
@caisq Well, I am expressing an interest to contribute. Honestly, I have a bit of a free time at hand and looking around for something challenging. Couldn't hurt to try. But, I am a bit new to this repository - so any  and all help would be appreciated.

**Comment by caisq** [comment]
@aniket-chowdhury I'm going to propose a concrete goal. You can accept or differ, depending on your thoughts and interest.

The goal is to enable writing out weight histograms during `tf.LayersModel.fit()` calls, when the tf.node.tensorBoard callback is configure. See this example:
- https://github.com/tensorflow/tfjs-examples/blob/fb430042376431fba804a2495891100f6e2205d6/sentiment/train.js#L193

and this documentation:
- https://js.tensorflow.org/api_node/1.1.0/#node.tensorBoard

The tensorBoard callback currently supports only scalars (losses and metrics).

The Python code you case off of is:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks.py#L1844

The equivalent TypeScript code in the tensorflow.js code base is at:
https://github.com/tensorflow/tfjs-node/blob/master/tfjs-node/src/callbacks.ts

In order to write histogram summaries, you'll need to implement the Histogram op first. Follow examples from the Scalar op:
https://github.com/tensorflow/tfjs/blob/ba1862ba7b22c5be268eda01abea6d8b1a5a1387/tfjs-node/src/nodejs_kernel_backend.ts#L1911

Be sure to include unit tests and clear docs, as usual.

Let me know if you need any help.

**Comment by aniket-chowdhury** [comment]
@caisq Sounds interesting. I guess I will begin with the looking up previous implementation for Ops and then work on the Histogram op.

**Comment by WenheLI** [comment]
This thread seems to be inactive for a while, can I take this issue over?

**Comment by aniketbiprojit** [comment]
You can take over @WenheLI 

This issue went beyond my scope.

**Comment by gaikwadrahul8** [comment]
Hi, @thes3m 

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFJs team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TFJs [version](https://github.com/tensorflow/tfjs/releases/tag/tfjs-v4.10.0) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tfjs/blob/master/README.md) to stay up to date with the latest developments which are happening in the Tensorflow.js space.

Thank you for your support and cooperation.
